hivemq-edge/src/main/java/com/hivemq/protocols/ProtocolAdapterManager.java

```diff
@@ -118,15 +117,13 @@ public ProtocolAdapterManager(
                 protocolAdapterWritingService.writingEnabled()));
     }
 
-    private synchronized @NotNull ProtocolAdapterWrapper createAdapterInternal(
+    private @NotNull ProtocolAdapterWrapper createAdapterInternal(
```

Comment:
Assuming 2 threads are calling this method at the same time with the same adapter ID,

- T1 calls `protocolAdapters.get(config.getAdapterId())` and gets `null`.
- T2 calls `protocolAdapters.get(config.getAdapterId())` and gets `null`.
- T1 calls `protocolAdapterFactory.createAdapter()`.
- T2 calls `protocolAdapterFactory.createAdapter()`.
- T1 calls `protocolAdapters.put()`.
- T2 calls `protocolAdapters.put()`.

Will that cause any problems?

Reply:
You are right. Wanted to fix this in a separate ticket but in the end it bothered me too much :)
Solved.